packages/gatsby/src/redux/index.ts

```diff
@@ -76,19 +76,19 @@ const multi: Middleware<IMultiDispatch> = ({ dispatch }) => next => (
 ): ActionsUnion | Array<ActionsUnion> =>
   Array.isArray(action) ? action.filter(Boolean).map(dispatch) : next(action)
 
+export type GatsbyReduxStore = Store<IGatsbyState> & {
+  dispatch: ThunkDispatch<IGatsbyState, any, ActionsUnion> & IMultiDispatch
+}
+
 // We're using the inferred type here because manually typing it would be very complicated
 // and error-prone. Instead we'll make use of the createStore return value, and export that type.
-// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
-export const configureStore = (
-  initialState: IGatsbyState
-): Store<IGatsbyState> =>
+export const configureStore = (initialState: IGatsbyState): GatsbyReduxStore =>
```

Comment:
I think we should remove comments above this line

Reply:
Done in https://github.com/gatsbyjs/gatsby/pull/32282